lib/target.js

```diff
@@ -369,23 +370,23 @@ class Target {
       this.packagerOptions['app-bundle-id'] += `.${this.channel}`;
     }
 
-    this.osx_dmg_label = this.osx_dmg_filename = `${this.productName}.dmg`;
-    this.osx_zip_label = this.osx_zip_filename = `${this.productName}.zip`;
+    this.osx_dmg_label = this.osx_dmg_filename = `${this.truncatedProductName}.dmg`;
+    this.osx_zip_label = this.osx_zip_filename = `${this.truncatedProductName}.zip`;
 
     this.assets = [
       {
         name: `${this.id}-${this.version}-${this.platform}-${this.arch}.dmg`,
-        path: this.dest(`${this.productName}.dmg`)
+        path: this.dest(`${this.truncatedProductName}.dmg`)
       },
       {
         name: `${this.id}-${this.version}-${this.platform}-${this.arch}.zip`,
-        path: this.dest(`${this.productName}.zip`)
+        path: this.dest(`${this.truncatedProductName}.zip`)
       }
     ];
 
     this.installerOptions = {
-      dmgPath: this.dest(`${this.productName}.dmg`),
-      title: this.productName,
+      dmgPath: this.dest(`${this.truncatedProductName}.dmg`),
+      title: this.truncatedProductName, // actually names the dmg
```

Comment:
I was under the assumption it's the title that's only causing the failure, and that the actual filename didn't need to be truncated since OSX can have longer filenames than 25 chars... Was this not the case? If it is the case - I think it's only the title that needs the truncated name.